subxt/Cargo.toml

```diff
@@ -15,31 +15,49 @@ description = "Submit extrinsics (transactions) to a substrate node via RPC"
 keywords = ["parity", "substrate", "blockchain"]
 
 [features]
-default = ["jsonrpsee-ws", "substrate-compat"]
+# For dev and documentation reasons we enable more features than are often desired.
+# it's recommended to use `--no-default-features` and then select what you need.
+default = ["jsonrpsee", "native", "substrate-compat"]
 
-# Activate this feature to pull in extra Substrate dependencies which make it
-# possible to provide a proper extrinsic Signer implementation (PairSigner).
+# Enable this for native (ie non web/wasm builds).
+# Exactly 1 of "web" and "native" is expected.
+native = [
+    "jsonrpsee?/async-client",
+    "jsonrpsee?/client-ws-transport"
+]
+
+# Enable this for web/wasm builds.
+# Exactly 1 of "web" and "native" is expected.
+web = [
+    "jsonrpsee?/async-wasm-client",
+    "jsonrpsee?/client-web-transport",
+    "getrandom/js"
+]
+
+# Enable this to use jsonrpsee (allowing for example `OnlineClient::from_url`).
+jsonrpsee = [
+    "dep:jsonrpsee"
+]
+
+# Enable this to pull in extra Substrate dependencies which make it possible to
+# use the `sp_core::crypto::Pair` Signer implementation, as well as adding some
+# `From` impls for types like `AccountId32`. Cannot be used with "web".
 substrate-compat = [
```

Comment:
so, `features = ["jsonrpsee", "native"]` should make it work but not bring in sp-core and all these deps, correct? 

This will be great

Reply:
Yup exactly :)

The only reason I left `substrate_compat` in the default deps iirc was so that doc examples (well, one doc example iirc, and maybe a couple of tests or something) would work. Probably need one more go-over on these features to sort out niggles like that because would be nicer not to have it by default at all!